pyciemss/integration_utils/observation.py

```diff
@@ -1,8 +1,44 @@
+from typing import Dict, Tuple
+
+import pandas as pd
+import torch
+
 from pyciemss.observation import NoiseModel, NormalNoiseModel
 
 _STR_TO_OBSERVATION = {"normal": NormalNoiseModel}
 
 
+def load_data(
+    path: str, data_mapping: Dict[str, str] = {}
+) -> Tuple[torch.Tensor, Dict[str, torch.Tensor]]:
+    """
+    Load data from a CSV file.
+
+    - path: path to the CSV file
+    - data_mapping: A mapping from column names in the data file to state variable names in the model.
+        - keys: str name of column in dataset
+        - values: str name of state/observable in model
+    - If not provided, we will assume that the column names in the data file match the state variable names.
+    """
+
+    df = pd.read_csv(path)
+
+    data_timepoints = torch.tensor(df["Timestamp"].values, dtype=torch.float32)
+    data = {}
+
+    for col in df.columns:
+        if col == "Timestamp":
+            continue
+
+        if col in data_mapping:
+            data[data_mapping[col]] = torch.tensor(df[col].values, dtype=torch.float32)
+        else:
+            data[col] = torch.tensor(df[col].values, dtype=torch.float32)
+        # TODO: address missing data
```

Comment:
Create a new PR for missing data, and also typecheck if a column is int, str, obj, or float